tools/release-plugin.sh

```diff
@@ -99,41 +125,43 @@ function check_ver {
 	return 0
 }
 
-if ! check_ver "$2"; then
-	die "Please specify a valid version number."
-fi
-VERSION="$2"
+for PLUGIN in "${!PROJECTS[@]}"; do
+	if ! check_ver "${PROJECTS[$PLUGIN]}" "$PLUGIN"; then
+		die "Please specify a valid version number."
+	fi
+	echo "Releasing $PLUGIN ${PROJECTS[$PLUGIN]}"
+done
 
-# Bail if we're passing an alpha or beta flag but not including it in the version number and vice versa.
-VERSION_AB=
-if VERSION_AB="$(grep -o '-a\|-beta' <<< "$VERSION")"; then
-	VERSION_AB="-${VERSION_AB:1:1}"
-fi
+proceed_p "" "Proceed releasing above projects?"
 
-if [[ -n "$ALPHABETA" ]]; then
-	if [[ "$VERSION_AB" != "$ALPHABETA" || ! "$VERSION_AB" ]]; then
-		die "$ALPHABETA passed to script, but version number $VERSION does not contain the corresponding flag."
-	fi
-elif [[ -n "$VERSION_AB" && -z "$ALPHABETA" ]]; then
-	die "$VERSION contains alpha or beta flag, but corresponding flag was not passed to the script, i.e. tools/release-plugin.sh -b plugins/jetpack 11.6-beta."
-fi
 
-proceed_p "Releasing $PROJECT $VERSION" "Proceed?"
+# Get the release branches for the projects.
+declare -A PREFIXES
 
-# Check if a remote branch for the release branch exits and ask to delete it if it does.
-PREFIX=$(jq -r '.extra["release-branch-prefix"] // ""' "$BASE"/projects/"$PROJECT"/composer.json)
-RELEASE_BRANCH=
-if [[ -n "$PREFIX" ]]; then
-	RELEASE_BRANCH="$PREFIX/branch-${VERSION%%-*}"
-else
-	die "No release branch prefix found for $PROJECT, aborting."
+# If we're releasing Jetpack, we need to set that prefix first.
+if [[ "${!PROJECTS[*]}" =~ "plugins/jetpack" ]]; then
+	PREFIXES["$(jq -r '.extra["release-branch-prefix"] // ""' "$BASE"/projects/plugins/jetpack/composer.json)"]=$(jq -r '.version' "$BASE"/projects/plugins/jetpack/composer.json)
 fi
 
-REMOTE_BRANCH="$(git ls-remote origin "$RELEASE_BRANCH")"
-if [[ -n "$REMOTE_BRANCH" ]]; then
-	proceed_p "Existing release branch $RELEASE_BRANCH found." "Delete it before continuing?"
-	git push origin --delete "$RELEASE_BRANCH"
-fi
+for SLUG in "${!PROJECTS[@]}"; do
+	PREFIX=$(jq -r '.extra["release-branch-prefix"] // ""' "$BASE"/projects/"$SLUG"/composer.json)
+	if [[ -n "$PREFIX" ]] && [[ ! "${PREFIXES[*]}" =~ $PREFIX ]]; then
```

Comment:
Same as line 83 again. Plus that will avoid trouble if `$PREFIX` ever happens to contain any regex special characters for some reason.

Also, do like `[[ test1 && test2 ]]` instead of `[[ test1 ]] && [[ test2 ]]`.

Reply:
Updated in fba64bd72e3ba38503bf17fa397e3f101bb053a4